src/ducks/recurrence/queries.js

```diff
@@ -20,7 +20,10 @@ export const bundleTransactionsQueryConn = ({ bundle }) => {
     query: () => {
       const initialQDef = queryRecurrenceTransactions(bundle)
       const qDef = initialQDef
-        .sortBy([{ date: 'desc' }])
+        .sortBy([
+          { 'relationships.recurrence.data._id': 'desc' },
+          { date: 'desc' }
+        ])
```

Comment:
From https://docs.cozy.io/en/tutorials/data/queries/#sort-with-cozy-client 

> ⚠️ If the fields involved in the sortBy are not indexed, this will force CouchDB to make the sort in memory: this can be acceptable if the query returns few documents, but it is not efficient for large queries.

You should create an index ;)

Reply:
The index is automatically created by cozy-client. https://github.com/cozy/cozy-client/blob/ea768adecaf0dd1c52e9876bf68e86d099579a9b/packages/cozy-stack-client/src/DocumentCollection.js#L339

Reply:
Yep also here https://github.com/cozy/cozy-client/blob/master/packages/cozy-pouch-link/src/CozyPouchLink.js#L325

thanks